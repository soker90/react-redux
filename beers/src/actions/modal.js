import modalAction from '../constants/actions/modal'

export default {
  openModal: modalData => ({type: modalAction.OPEN_MODAL, modalData}),
  closeModal: () => ({type: modalAction.CLOSE_MODAL})
}
