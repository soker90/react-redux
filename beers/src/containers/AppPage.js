import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import App from '../components/App'

export function mapStateToProps(state) {
  return {modal: {...state.modal}}
}

export function mapDispatchToProps(dispatch) {
  return {modalAction: bindActionCreators(modalActions, dispatch)}
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
