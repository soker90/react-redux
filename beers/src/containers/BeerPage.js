import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import beerActions from '../actions/beer'
import Beer from '../components/Beer'

export function mapStateToProps (state) {
  return {...state.beers, ...state.modal}
}

export function mapDispatchToProps (dispatch) {
  return bindActionCreators({...beerActions, ...modalActions}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Beer)
