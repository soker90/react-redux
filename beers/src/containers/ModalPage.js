import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import Modal from '../components/Modal'

export function mapStateToProps (state) {
  return {...state.modal}
}

export function mapDispatchToProps (dispatch) {
  return bindActionCreators(modalActions, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Modal)
