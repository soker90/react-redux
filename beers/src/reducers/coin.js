import coinAction from '../constants/actions/coin'

export function initialState () {
  return {
    coinList: [],
    coin: {},
    loading: false,
    error: null
  }
}

function fetchCoin(state) {
  return {
    ...state,
    coinList: [],
    loading: true,
    error: null
  }
}

function fetchCoinSuccess(state, {coins}) {
  return {
    ...state,
    coinList: coins.coins,
    loading: false,
    error: null
  }
}

function addCoin(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function addCoinSuccess(state, {coin}) {
  return {
    ...state,
    coinList: [coin, ...state.coinList],
    loading: false,
    error: null
  }
}

function removeCoin(state) {
  return {
    ...state,
    loading: true,
    error: null
  }
}

function removeCoinSuccess(state, {coinId}) {
  return {
    ...state,
    coinList: state.coinList.filter((coin) => coin.id !== coinId.coinId),
    loading: false,
    error: null
  }
}

function coinError(state, {error}) {
  return {
    ...state,
    loading: false,
    error: error
  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case coinAction.FETCH_COIN:
      return fetchCoin(state, action)
    case coinAction.FETCH_COIN_SUCCESS:
      return fetchCoinSuccess(state, action)
    case coinAction.ADD_COIN:
      return addCoin(state, action)
    case coinAction.ADD_COIN_SUCCESS:
      return addCoinSuccess(state, action)
    case coinAction.REMOVE_COIN:
      return removeCoin(state, action)
    case coinAction.REMOVE_COIN_SUCCESS:
      return removeCoinSuccess(state, action)
    case coinAction.COIN_ERROR:
      return coinError(state, action)
    default:
      return state
  }
}
