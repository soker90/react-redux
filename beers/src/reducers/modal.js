import modalAction from '../constants/actions/modal'

export function initialState () {
  return {
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}
  }
}

function openModal(state, {modalData}) {
  return {
    ...state,
    open: true,
    title: modalData.title,
    body: modalData.body,
    callback: modalData.callback ? modalData.callback : null,
    extraParams: modalData.extraParams ? modalData.extraParams : {}
  }
}

function closeModal(state) {
  return {
    ...state,
    open: false,
    title: '',
    body: '',
    callback: null,
    extraParams: {}

  }
}

export default function (state = initialState(), action) {
  switch (action.type) {
    case modalAction.OPEN_MODAL:
      return openModal(state, action)
    case modalAction.CLOSE_MODAL:
      return closeModal(state, action)
    default:
      return state
  }
}
