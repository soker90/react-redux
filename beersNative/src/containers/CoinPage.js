import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import modalActions from '../actions/modal'
import coinActions from '../actions/coin'
import Coin from '../components/Coin'

export function mapStateToProps (state) {
  return {...state.coins, ...state.modal}
}

export function mapDispatchToProps (dispatch) {
  return bindActionCreators({...coinActions, ...modalActions}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Coin)
