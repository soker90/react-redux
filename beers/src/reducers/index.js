import { combineReducers } from 'redux'
import beers from './beer'
import modal from './modal'
import coins from './coin'
import { reducer as formReducer } from 'redux-form'

const rootReducer = combineReducers({
  beers,
  modal,
  coins,
  form: formReducer
})

export default rootReducer
