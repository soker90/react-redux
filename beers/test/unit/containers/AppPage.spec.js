import {mapStateToProps, mapDispatchToProps} from '../../../src/containers/AppPage'
import { initialState } from '../../../src/reducers/modal'

describe('Containers', () => {
  test('should map AppPage state to props', () => {
    const state = {
      modal: initialState()
    }

    expect(mapStateToProps(state)).toMatchSnapshot()
  })

  test('should map AppPage actions to props', () => {
    expect(mapDispatchToProps()).toMatchSnapshot()
  })
})
