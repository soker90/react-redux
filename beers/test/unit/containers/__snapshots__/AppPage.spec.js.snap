exports[`Containers should map AppPage actions to props 1`] = `
Object {
  "modalAction": Object {
    "closeModal": [Function],
    "openModal": [Function],
  },
}
`;

exports[`Containers should map AppPage state to props 1`] = `
Object {
  "modal": Object {
    "body": "",
    "callback": null,
    "extraParams": Object {},
    "open": false,
    "title": "",
  },
}
`;
